### PR TITLE
implement html retriever + test

### PR DESCRIPTION
**Pull request by traceylui**
Not sure why @NoArgsConstructor has to be there for it to work, but it does retrieve the html

### CODE REVIEW

File: src/main/java/holidaybot/HolidaybotConfig.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package holidaybot;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+
+@Data
+@AllArgsConstructor
```
Comment (rwblickhan) [inline src/main/java/holidaybot/HolidaybotConfig.java]: `@Data` should already include `@RequiredArgsConstructor` so I think you can get rid of `@AllArgsConstructor` (the tests still run without it).

File: src/main/java/holidaybot/HolidaybotConfigManager.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.File;
+import java.io.IOException;
+
+public class HolidaybotConfigManager {
+
+    private static HolidaybotConfig configs;
+
```
Comment (rwblickhan) [inline src/main/java/holidaybot/HolidaybotConfigManager.java]: Mind getting rid of the extra line 😛

File: src/main/java/holidaybot/HolidaybotConfig.java (hunk 1 above)
Comment (traceylui) [reply src/main/java/holidaybot/HolidaybotConfig.java]: aight, changed!

File: src/main/java/holidaybot/HolidaybotConfigManager.java (hunk 2 above)
Comment (traceylui) [reply src/main/java/holidaybot/HolidaybotConfigManager.java]: why are you against whitespace!!!! 
changed tho

File: src/main/java/holidaybot/HtmlRetriever.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+        try (
+            CloseableHttpClient client = HttpClients.createDefault()) {
+            HttpGet httpGet = new HttpGet(configs.getHolidayUrl());
+            ResponseHandler<String> handler = new BasicResponseHandler();
+
+            HttpResponse response = client.execute(httpGet);
+
+            if (200 != response.getStatusLine().getStatusCode()) {
+                throw new HttpException("Unable to connect and retrieve holiday information :(");
```
Comment (jiahuahl) [inline src/main/java/holidaybot/HtmlRetriever.java]: Could you print the error response too, if it exists

File: src/main/java/holidaybot/HolidaybotConfigManager.java (hunk 2 above)
Comment (rwblickhan) [reply src/main/java/holidaybot/HolidaybotConfigManager.java]: white space is the devil

File: src/main/java/holidaybot/HolidaybotConfig.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package holidaybot;
+
+import lombok.AllArgsConstructor;
```
Comment (rwblickhan) [inline src/main/java/holidaybot/HolidaybotConfig.java]: You can get rid of this import.

**Pull request merged**